File: src/main/java/ch/qligier/jetbrains/fsh/parser/FshTokenType.java
package ch.qligier.jetbrains.fsh.parser;

import ch.qligier.jetbrains.fsh.language.FshLanguage;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

/**
 * jetbrains-language-fsh
 *
 * @author Quentin Ligier
 **/
public class FshTokenType extends IElementType {

    /**
     * Creates and registers a new element type for the specified language.
     *
     * @param debugName the name of the element type, used for debugging purposes.
     */
    public FshTokenType(@NonNls @NotNull final String debugName) {
        super(debugName, FshLanguage.INSTANCE);
    }
}

File: src/main/kotlin/ch/qligier/jetbrains/plugin/fss/fsh/language/psi/FshTokenType.kt
/*
 * Copyright 2026 Quentin Ligier. Use of this source code is governed by the MIT license.
 */

package ch.qligier.jetbrains.plugin.fss.fsh.language.psi

import ch.qligier.jetbrains.plugin.fss.fsh.FshLanguage
import com.intellij.psi.tree.IElementType
import org.jetbrains.annotations.NonNls

class FshTokenType(debugName: @NonNls String) : IElementType(debugName, FshLanguage) {
    override fun toString(): String {
        return "FshTokenType." + super.toString()
    }
}

